### PR TITLE
Added showTerms feature flag

### PR DESCRIPTION
**Pull request by cocojoe**
**Description**

Added a `showTerms` feature flag to disable the display of the terms of service dialog. Displayed in the Sign Up Screen for Database connections.
Note: If `mustAcceptTerms` is enabled, terms will always be shown.

screenshot_1538661279

### CODE REVIEW

File: lib/src/main/java/com/auth0/android/lock/Lock.java [hunk 1]
```diff
@@ -567,6 +567,19 @@ public Builder setMustAcceptTerms(boolean mustAcceptTerms) {
             return this;
         }
 
+        /**
+         * Displays the Privacy Policy and Terms of Service dialog.
```
Comment (lbalmaceda) [inline lib/src/main/java/com/auth0/android/lock/Lock.java]: "Displays the Privacy Policy and Terms of Service footer on the Sign Up screen"

**Comment by cocojoe** [comment]
@lbalmaceda can you rephrase your question please, what do you mean into the custom fields sign up screen? Are you talking about how it looks with custom fields and/or how it looks if you set `initial screen to signup` ?

**Comment by cocojoe** [comment]
screenshot_1538666109
screenshot_1538666129

**Comment by lbalmaceda** [comment]
@cocojoe when your lock configuration has more than 3 fields the remaining fields are displayed in a separate screen, following the initial sign up screen. On that screen that I remember, the terms are not shown. With the changes on the code you added here it might be interesting checking that that use case still works with and without this new flag enabled

**Comment by cocojoe** [comment]
screenshot_1538743686
